### DISCUSSION TITLE
how to DefaultPredictor reading multipe images

### DISCUSSION THREAD

**Discussion by mrmoonshine24** [facebookresearch/detectron2#3894] [Q&A]
Dear all,

I have trained a model which works succesfully. However when I am using defaultpredictor, I can only detect 1 images at the time. Which is fine but I was wondering how I can create something that will work for multiple images in the same folder. I dont need batch interference since it is not big enough, but to change the code for the predictor which images to give a result for is abit to much work. 

Could someone help me out?

Thanks!

**Comment by zhanghang1989** [comment]
It sounds like using a for loop should solve this problem.
